packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -642,12 +655,19 @@ export async function emitCode(context: vscode.ExtensionContext, uri: vscode.Uri
           }
         } else {
           resolve([...existingEmittersSelector.selectedItems]);
-          existingEmittersSelector.dispose();
+          existingEmittersSelector.hide();
+        }
+      });
+
+      existingEmittersSelector.onDidHide(() => {
```

Comment:
Does it timeout or user has to close it?

Reply:
the event will be triggered as long as the quickpick is hidden from the UI. i.e.  hide() method is called, user ESC the quickpick, some other UI makes the quickpick hidden and so on.